frontends/PyCDE/src/pycde/value.py

```diff
@@ -384,36 +374,39 @@ def __ne__(self, other):
 
   def __and__(self, other):
     from .dialects import comb
-    return self.__exec_signless_binop__(other, comb.AndOp, "and")
+    return self.__exec_signless_binop_nocast__(other, comb.AndOp, "&", "and")
 
   def __or__(self, other):
     from .dialects import comb
-    return self.__exec_signless_binop__(other, comb.OrOp, "or")
+    return self.__exec_signless_binop_nocast__(other, comb.OrOp, "|", "or")
 
   def __xor__(self, other):
     from .dialects import comb
-    return self.__exec_signless_binop__(other, comb.XorOp, "xor")
+    return self.__exec_signless_binop_nocast__(other, comb.XorOp, "^", "xor")
 
   def __invert__(self):
     from .pycde_types import types
-    ret = self.as_int() ^ types.int(self.type.width)(-1)
+    ret = self ^ types.int(self.type.width)(-1)
     if self.name is not None:
-      ret.name = f"neg_{self.name}"
+      ret.name = f"inv_{self.name}"
     return ret
 
+
+class IntValue(BitVectorValue):
+  # hwarith icmp predicate constants.
+  _ICMP_PRED_EQ = 0
+  _ICMP_PRED_NE = 1
```

Comment:
Could these be exposed through the python bindings? or do you think that would be overkill?

Reply:
Yeah, they should probably go in the hwarith ext module. It doesn't seem like td enums currently make it into the auto generated python.